services/productService.js

```diff
@@ -0,0 +1,29 @@
+const productDao = require("../models/productDao.js");
+
+const allProducts = async (page, pageSize, category, sub, type, sort) => {
+  let start = 0;
+  if (page <= 0) {
+    page = 1;
+  } else {
+    start = (page - 1) * pageSize;
+  }
```

Comment:
limit과 offset에 대해 클라이언트가 요청하는 데이터가 없다면 매개변수에서 deafultValue를 설정해줄 수 있습니다. 
만약 데이터가 들어온다면 deafultValue 대신 클라이언트가 요청한 데이터로 치환되어 각 레이어에 전달됩니다.

```suggestion
const allProducts = async (category, subCategory, type, sort, limit=15, offset=0) => {
```